Objects/weakrefobject.c

```diff
@@ -894,13 +894,35 @@ PyWeakref_NewProxy(PyObject *ob, PyObject *callback)
 }
 
 
+int
+PyWeakref_GetRef(PyObject *ref, PyObject **pobj)
+{
+    if (ref == NULL) {
+        *pobj = NULL;
+        PyErr_BadInternalCall();
+        return -1;
+    }
+    if (!PyWeakref_Check(ref)) {
+        *pobj = NULL;
+        PyErr_SetString(PyExc_TypeError, "expected a weakref");
+        return -1;
+    }
+    *pobj = _PyWeakref_GET_REF(ref);
+    return 0;
+}
+
+
 PyObject *
 PyWeakref_GetObject(PyObject *ref)
 {
-    if (ref == NULL || !PyWeakref_Check(ref)) {
+    if (ref == NULL) {
         PyErr_BadInternalCall();
         return NULL;
     }
+    if (!PyWeakref_Check(ref)) {
+        PyErr_SetString(PyExc_TypeError, "expected a weakref");
+        return NULL;
+    }
```

Comment:
I think this behaviour change should get it its own PR and NEWS item.

Reply:
I thought that it would be nice to make the two functions consistent, but I'm not motivated to write a dedicated PR just for that. I prefer to restrict this PR to PyWeakref_GetRef(), so I revert it.

Reply:
Sounds good!